resharper/resharper-unity/src/CSharp/Daemon/Stages/Analysis/PropertyDrawerOnGUIBaseProblemAnalyzer.cs

```diff
@@ -0,0 +1,85 @@
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Errors;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Stages.Dispatcher;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Impl.Reflection2;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Stages.Analysis
+{
+    [ElementProblemAnalyzer(typeof(IInvocationExpression), HighlightingTypes =
+        new[] {typeof(PropertyDrawerOnGUIBaseWarning)})]
+    public class PropertyDrawerOnGUIBaseProblemAnalyzer : UnityElementProblemAnalyzer<IInvocationExpression>
+    {
+        public PropertyDrawerOnGUIBaseProblemAnalyzer(UnityApi unityApi)
+            : base(unityApi)
+        {
+        }
+
+        protected override void Analyze(IInvocationExpression expression, ElementProblemAnalyzerData data,
+            IHighlightingConsumer consumer)
+        {
+            if (IsOnGUIBaseCall(expression) &&
+                IsInsidePropertyDrawer(expression) &&
+                IsInsideOnGUI(expression))
```

Comment:
Why only check inside OnGUI? Isn't every call to `PropertyDrawer.OnGUI` bad?

We could also probably get rid of the check for base type deriving from `PropertyDrawer` too, because we're checking for the `base.` qualifier, and checking that `OnGUI` is derived on `PropertyDrawer`